lacci/lib/shoes/drawable.rb

```diff
@@ -242,15 +242,17 @@ def shoes_style_name?(name)
       # not kept long, and used up when used once.
 
       def with_current_app(app)
-        old_app = Thread.current[:shoes_app]
-        Thread.current[:shoes_app] = app
-        yield
-      ensure
-        Thread.current[:shoes_app] = old_app
+        old_cur_app = @current_app
+        @current_app = app
+        ret = yield
+        @current_app = old_cur_app
```

Comment:
This looks good to me. You *could* use an ensure when setting @current_app to old_cur_app, but I don't remember if we were before.